core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala

```diff
@@ -414,6 +415,99 @@ class ExternalAppendOnlyMapSuite extends SparkFunSuite with LocalSparkContext {
     sc.stop()
   }
 
+  test("spill during iteration") {
```

Comment:
I understand what this test want to do. But it seems code without this PR could also pass it if everything goes normally. And I know it's a little hard to reflect the change by unit test. Also, I'd prefer to leave some comments to explain the potential memory leak in source code above.

Reply:
This test was written BEFORE the actual fix and it did fail up untill the fix was in place. I do agree it's a bit clumsy and potential future changes may break the original intention of the test. I've referred a potential testing approach (currently limited to scala's source code) which couldn't be (easily) applied to this code base so I made a best effort to test this.
I agree this needs better documentation, I'll start be referring the issue in the test's name and will also add comments to the code.